### DISCUSSION TITLE
Warning With Code Suggestion Should Stay Open Longer or Until Acknowledged

### DISCUSSION THREAD

**Discussion by veaux** [forcedotcom/cli#2760] [Ideas]
The warning and code suggestion when a code is incorrectly input or a deprecated code is entered disappears too quickly. The code suggestion which is followed by [y/n] should either remain on the screen much longer until it disappears OR remain indefinitely until acted upon by the user. It is a good feature just disappears to fast.

```
'sf org ope'
»   Warning: force org ope is not a sf command.
Did you mean force org open? [y/n]:
```
The above usually disappears from the command line much before I'm able to act on it since I initially felt like the command was correct so disengaged from screen for a period of time until the command was completed. This is obviously just 1 simple example but the same holds true for any sf command entered that results in a code suggestion.

**Comment by mdonnalley** [comment]
We have a timeout on the prompt because we never want to cause customer's CI pipelines to hang indefinitely because of typo.

But we'd consider adding an environment variable that would override the default 10 second timeout - is that something you'd be interested in?

**Reply by daveespo** [reply]
But doesn't the non-TTY environment skip the timeout altogether (as was broken and then fixed on #2739 ) 

It seems like a longer, or maybe infinite, timeout on a TTY environment isn't as scary as it sounds?

**Comment by veaux** [comment]
The variable seems like that would work esp given that it doesn't seem
helpful on CI side, only UI.
